src/Server.js

```diff
@@ -0,0 +1,229 @@
+    }
+
+    loadCommands() {
+        // cmdline options
+        commander
+            .version(pkg.version)
+            .option('-s, --save [time]', 'Number of seconds between auto-save ticks [10]', 10)
+            .option('-r, --respawn [time]', 'Number of minutes between respawn ticks [120]', 120)
+            .option('-p, --port [portNumber]', 'Port to host telnet server [23]', 23)
+            .option('-v, --verbose', 'Verbose console logging.', true)
+            .parse(process.argv);
+
+        process.stdin.setEncoding('utf8');
+        process.stdin.resume();
+        process.stdin.on('data', data => {
+            data = data.trim();
+            const command = data.split(' ')[0];
+            const args = data.split(' ').slice(1).join(' ');
+
+            switch (command) {
+                case "save":
+                    this.save(args);
+                    break;
+                case "reload":
+                    this.reload(args);
+                    break;
+                case "restart":
+                    this.restart(args);
+                    break;
+                case "shutdown":
+                    this.shutdown(args);
+                    break;
+                default:
+                    console.log("That's not a real command...");
+                    return;
+                    break;
+            }
+        });
+    }
+
+    save(callback) {
+        this.GameState.PlayerManager.saveAll();
+        if (callback) {
+            callback();
+        }
+    }
+
+    reload(args) {
+        this.serverOperation('SERVER RELOAD', () => {
+            util.log("Reloading...");
+            this.save(() => this.init(false));
+        }, args);
+    }
+
+    restart(args) {
+        this.serverOperation('SERVER RESTART', () => {
+            util.log("Restarting...");
+            this.save();
+            this.server.emit('shutdown');
+            this.server.close();
+            this.init(true);
+        }, args);
+    }
+
+    shutdown(args) {
+        this.serverOperation('SERVER SHUTDOWN', () => {
+            util.log("Shutting down...");
+            this.save();
+            this.server.emit('shutdown');
+            this.server.close();
+        }, args);
+    }
+
+    serverOperation(warningStr, operation, args) {
+        args = args ? args.split(' ') : [];
+        const shouldWarn = args[0] && args[0] === 'warn';
+
+        let time = args[0] ? parseInt(args[shouldWarn ? 1 : 0], 10) : 0;
+
+        if (time && time < 20) {
+            console.log("Gotta give the players a bit longer than that, might as well do it instantly...");
+            console.log("Doing nothing.");
+            return;
+        }
+
+        time = time ? time * 1000 : 0;
+
+        global.countdown = time;
+
+        if (shouldWarn) {
+            const warnPlayers = interval => {
+                this.GameState.PlayerManager.players.forEach(p => Broadcast.sayAt(p, `${warningStr} ${global.countdown/ 1000} SECONDS`));
```

Comment:
Should be

    Broadcast.sayAt(this.GameState.PlayerManager, `${warningStr} ${global.countdown/1000} SECONDS`);